src/main/kotlin/nl/connectplay/scoreplay/controllers/ScoreController.kt

```diff
@@ -0,0 +1,108 @@
+package nl.connectplay.scoreplay.controllers
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.request.*
+import nl.connectplay.scoreplay.abstraction.data.ScoreRepository
+import nl.connectplay.scoreplay.models.dto.score.CreateScoreDto
+import nl.connectplay.scoreplay.models.dto.score.UpdateScoreDto
+import nl.connectplay.scoreplay.utilities.getLimitQueryParameter
+import nl.connectplay.scoreplay.utilities.getOffsetQueryParameter
+import java.sql.SQLException
+import java.util.UUID
+
+class ScoreController(
+    private val scoreRepository: ScoreRepository) {
+
+    // Get /Scores
+    suspend fun handleListAsync(call: ApplicationCall) {
+        val limit = call.request.getLimitQueryParameter()
+        val offset = call.request.getOffsetQueryParameter()
+
+
+        try {
+            val scores = scoreRepository.getScoresAsync(limit, offset)
+                ?: return call.respond(HttpStatusCode.NoContent)
+
+            call.respond(HttpStatusCode.OK,scores)
+        }catch (e: SQLException) {
+            call.application.environment.log.error("No scores found")
+            call.respond(HttpStatusCode.InternalServerError)
+        }
+    }
+
+    //GET /Scored/ID
+    suspend fun handleOneAsync(call: ApplicationCall) {
+        val scoreIdParam = call.request.queryParameters["scoreId"] ?:
+        return call.respond(HttpStatusCode.BadRequest, "Missing score id")
+
+        val scoreId = try {
+            UUID.fromString(scoreIdParam)
+        } catch (e: SQLException) {
+            return call.respond(HttpStatusCode.BadRequest, "Invalid id")
+        }
+
+        try {
+            val score = scoreRepository.getScoreByIdAsync(scoreId)
+                ?: return call.respond(HttpStatusCode.NoContent, "Score not found")
+
+            call.respond(HttpStatusCode.OK, score)
+        } catch (e: SQLException) {
+            call.application.environment.log.error("No score found")
```

Comment:
log de exception ook.
```suggestion
            call.application.environment.log.error("DB error while looking up score $scoreId", e)
```